lib/libc/newlib/include/stdint.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_LIB_LIBC_NEWLIB_INCLUDE_STDINT_H_
+#define ZEPHYR_LIB_LIBC_NEWLIB_INCLUDE_STDINT_H_
+
+/* Work around -ffreestanding absence of defines required to support
+ * PRI.64 macros in <inttypes.h> by including the newlib header that
+ * provides the flag macros.
+ */
+
+#include <newlib.h>
+
+#ifdef __NEWLIB__
+/* Has this header.  Older versions do it in <stdint.h>. */
+#include <sys/_stdint.h>
+#endif /* __NEWLIB__ */
+
+/* This should work on GCC and clang. */
+#if ((__GNUC__ - 0) > 0) || __has_include_next(<stdint.h>)
```

Comment:
Was effect of this `__has_include_next` thingy considered on other toolchains? I personally never saw it before, and a quick google-up show a usage wrapped in `#ifdef __has_include_next`: https://www.gnu.org/software/gcc/gcc-5/changes.html

Reply:
I found it in [clang](http://clang.llvm.org/docs/LanguageExtensions.html#has-include-next) when checking whether that supported `#include_next`.  I wasn't aware that it was also supported in GNUC.

After further review I've decided the protection stuff was unnecessarily complex, and replaced it with a bare use of `#include_next` and a comment saying what needed to be done if some toolchain doesn't support that capability.